### PR TITLE
Deprecate class extension

### PR DESCRIPTION
**Pull request by greg0ire**
This is to prepare major signature changes that will come when dropping
PHP < 7 support. These changes are actually already done on the master
branch. This should make the transition smoother.

see https://github.com/sonata-project/cache/pull/65

I am targetting this branch, because this is BC.

See https://github.com/sonata-project/GoogleAuthenticator/projects/1#card-1828423

## Changelog

```markdown
### Deprecated
- Extending any class defined in the package is deprecated
```

### CODE REVIEW

File: src/FixedBitNotation.php [hunk 1]
```diff
@@ -39,6 +39,16 @@ class FixedBitNotation
      */
     public function __construct($bitsPerCharacter, $chars = null, $rightPadFinalBits = false, $padFinalGroup = false, $padCharacter = '=')
     {
+        /* NEXT_MAJOR:
+          - remove this block
+          - make this class final
+          - and properties private */
+        if (__CLASS__ !== get_class($this)) {
```
Comment (jordisala1991) [inline src/FixedBitNotation.php]: nice trick!

Comment (greg0ire) [inline src/FixedBitNotation.php]: Thanks :)

**Comment by greg0ire** [comment]
I mean signature changes.

**Comment by jordisala1991** [comment]
Oh, okay.

**Comment by greg0ire** [comment]
> Even if my approvals are weaker, I approve this.

Lol. Please do. And @sonata-project/contributors , please take this approval into account when merging.

**Comment by OskarStark** [comment]
Thank you @greg0ire 👍

**Pull request merged**